lib/archethic.ex

```diff
@@ -74,25 +73,50 @@ defmodule Archethic do
     end
   end
 
-  defp forward_transaction(
-         tx,
-         welcome_node_key,
-         nodes \\ P2P.authorized_and_available_nodes()
-         |> Enum.filter(&Node.locally_available?/1)
-         |> P2P.nearest_nodes()
-       )
+  @spec forward_transaction(tx :: Transaction.t(), welcome_node_key :: Crypto.key()) :: :ok
+  defp forward_transaction(tx, welcome_node_key) do
+    %Node{network_patch: welcome_node_patch} = P2P.get_node_info!(welcome_node_key)
+
+    nodes =
+      P2P.authorized_and_available_nodes()
+      |> Enum.reject(&(&1.first_public_key == welcome_node_key))
+      |> Enum.sort_by(& &1.first_public_key)
```

Comment:
is this `sort_by` useful? look like the `nearest_nodes` is resorting it right after?

Reply:
This sort is usefull when there is multiple nodes in the same patch. We need to have the same order of node within the same patch, while the nearest_nodes function does not ensure the sort inside a patch